src/app/analyze/stack/stack-details/stack-details.component.ts

```diff
@@ -76,14 +252,12 @@ export class StackDetailsComponent implements OnInit {
         this.componentsWithTests = this.stackAnalysesData[0].metadata.components_with_tests;
         this.componentsWithDependencyLockFile = this.stackAnalysesData[0].metadata.components_with_dependency_lock_file;
         this.requiredEngines = this.stackAnalysesData[0].metadata.required_engines;
-        for (let key in this.requiredEngines) {
-          if (this.requiredEngines.hasOwnProperty(key)) {
-            this.requiredEnginesArr.push({ key: key, value: this.requiredEngines[key] });
-          }
+        for (var key in this.requiredEngines) {
+          this.requiredEnginesArr.push({ key: key, value: this.requiredEngines[key] });
         }
 
-        for (let i = 0; i < this.stackAnalysesData[0].components.length; i++) {
-          let myObj = Object.assign({}, this.stackAnalysesModel);
+        for (var i = 0; i < this.stackAnalysesData[0].components.length; i++) {
+          var myObj = Object.assign({}, this.stackAnalysesModel);
```

Comment:
It looks like you are manually doing a deep copy. If that is the case I would recommend using lodash instead of writing your own.